workspaces/announcements/.changeset/dull-yaks-hang.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage-community/plugin-announcements': patch
+---
+
+Re-fetching announcement details on route parameter change
```

Comment:
Let's update this to be verbose about how this fixes fetching the proper announcement details when route parameters change.

Reply:
@kurtaking I have added a more detailed comment.

Reply:
@drapalaa - ready to take the PR out of draft?